customer/src/main/java/no/unit/nva/customer/update/UpdateCustomerDoiHandler.java

```diff
@@ -58,46 +56,54 @@ protected List<MediaType> listSupportedMediaTypes() {
     }
 
     @Override
-    protected DoiAgentDto processInput(DoiAgentDto input, RequestInfo requestInfo, Context context)
+    protected String processInput(String input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
 
         authorizeDoiAgentChange(requestInfo);
 
         var customerId = getIdentifier(requestInfo);
-        var customer = customerService.getCustomer(customerId);
-        customer.setDoiAgent(input);
-        var result = customerService.updateCustomer(customerId, customer);
-
-        var secretsListFiltered = getSecretManagerDoiAgentsFiltered(customer.getId());
 
-        var doiSecret = new SecretManagerDoiAgentDao(customer.getId(), input);
-
-        var allSecretsJsonString =
-            Stream.concat(secretsListFiltered, Stream.of(doiSecret))
-                .map(SecretManagerDoiAgentDao::toString)
-                .collect(Collectors.joining(",", "[", "]"));
+        var persistedDoiAgent = updateSecretManager(customerId, input);
 
-        secretsWriter.updateSecretKey(
-            CUSTOMER_DOI_AGENT_SECRETS_NAME,
-            CUSTOMER_DOI_AGENT_SECRETS_NAME,
-            allSecretsJsonString);
+        updateCustomer(customerId, persistedDoiAgent);
```

Comment:
this comes after `updateSecretManager` ? It was the other way around before.

Reply:
Correct! Sharp eyes.
After implementing merge, the input value could return a merged (modified) entity that needs to be reflected in customer persist as well.